src/js/features/ingestion/lastIngestion.store.ts

```diff
@@ -0,0 +1,73 @@
+import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
+import axios from 'axios';
+import { katsuLastIngestionsUrl, gohanLastIngestionsUrl } from '@/constants/configConstants';
+import { printAPIError } from '@/utils/error.util';
+
+export interface DataTypeResponse {
+  count: number | null;
+  id: string;
+  label: string;
+  last_ingested: string | null;
+  queryable: boolean;
+}
+
+export type DataResponseArray = DataTypeResponse[];
+
+// Async thunks to fetch data from the two endpoints
+export const fetchKatsuData = createAsyncThunk('dataTypes/fetchKatsuData', (_, { rejectWithValue }) =>
+  axios
+    .get(katsuLastIngestionsUrl)
+    .then((res) => res.data)
+    .catch(printAPIError(rejectWithValue))
+);
+
+export const fetchGohanData = createAsyncThunk('dataTypes/fetchGohanData', (_, { rejectWithValue }) =>
+  axios
+    .get(gohanLastIngestionsUrl)
+    .then((res) => res.data)
+    .catch(printAPIError(rejectWithValue))
+);
+
+// Define the state structure
+export interface DataTypeState {
+  isFetchingData: boolean;
+  dataTypes: DataResponseArray;
+}
+
+// Initialize the state
+const initialDataTypeState: DataTypeState = {
+  isFetchingData: false,
+  dataTypes: [],
+};
+
+const reduceServiceDataTypes = (state: DataTypeState, { payload }: PayloadAction<DataResponseArray>) => {
+  const uniqueIds = new Set(state.dataTypes.map((data: DataTypeResponse) => data.id));
+  const newData = payload.filter((data: DataTypeResponse) => !uniqueIds.has(data.id));
+  state.dataTypes = [...state.dataTypes, ...newData];
```

Comment:
this means that the data type array order will depend on which service returns first, and will flip around randomly based on this from the user's perspective. it makes more sense to make the data types in the store an object in my opinion, and set

```ts
state.dataTypes = {
    ...state.dataTypes, 
    ...Object.fromEntries(payload.map((data: DataTypeResponse) => [data.id, data])),
};
```

and then, when rendering, use `Object.entries` and apply sorting based on the label.